internal/venice-common/src/main/java/com/linkedin/venice/stats/dimensions/VeniceRequestValidationOutcome.java

```diff
@@ -0,0 +1,15 @@
+package com.linkedin.venice.stats.dimensions;
+
+public enum VeniceRequestValidationOutcome {
+  VALID("valid"), INVALID_KEY_COUNT_LIMIT_EXCEEDED("invalid_key_count_limit_exceeded");
```

Comment:
1. Request vs response: It felt a bit weird to me to use response status codes for the requests that are not actually passed down. Does this concern hold any actual weight? WDYT
2. yes, my intention was to map more outcomes as and when I encounter it through this exercise.

Reply:
Hmm, IDK, TBH. I guess... we expect there to be a response from the router and the response should have some status code, regardless of whether the request was wrong and not passed at all to the server tier, or whether it was passed down to servers but then the server's response carried some error.

We may know that for some response status codes, the request to servers happens to have been short-circuited, but IDK if we would do anything differently based on that knowledge (maybe metrics reporting-wise we would have some kind of umbrella metric, or metric dimension, to capture all requests into either of these buckets).

I'm not sure what's ideal. I'm just thinking out loud since reading this file brought up these questions for me.

Reply:
Yes, I was under the similar confusion of whether to just use reponse context for the requests or whether trying to differentiate it makes better sense (went with the latter to start with). One other thing to take into consideration is not to create too many metrics or duplicates.
I like the idea of using response codes and separate metrics for the cases we are interested to track separately as well but probably not everything and if not needed we can retire the additional metrics at a later point in time (or the other way around: create as and when needed)

Reply:
I thought about this further. Currently we are recoding this metrics during request parsing and its either recorded as valid or bad. If I am to use the response code here, I can use 413 for bad (> max key count), but I don't have an equivalent for valid. This valid request can succeed or fail or return one of the response codes. Since we are asked to always provide a value to a dimension (Not Otel guideline: but for the pre-aggs to work properly), IMHO using response code will be confusing here.